src/compiler/parser.ts

```diff
@@ -298,26 +300,26 @@ module ts {
 
     function parsingContextErrors(context: ParsingContext): DiagnosticMessage {
         switch (context) {
-            case ParsingContext.SourceElements:         return Diagnostics.Declaration_or_statement_expected;

```

Comment:
Every now and then while typing in VS, the editor looses track of what node its editing when I write an end curly brace and reformats the whole file, its not always obvious when this happens and you're on line 4,000...